components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -268,7 +268,10 @@ func Run(options ...RunOption) {
 	// We keep the reaper until the bitter end because:
 	//   - it doesn't need graceful shutdown
 	//   - we want to do as much work as possible (SIGTERM'ing reparented processes during shutdown).
-	go reaper(terminatingReaper)
+	//   - in headless task we don't need reaper
+	if !cfg.isHeadless() {
```

Comment:
Good choice. Much preferable over a random `sleep` :)

Reply:
A more extensive comment would be awesome. "we can not because" reads more explicit than "don't need"